File: app/src/main/java/com/coinninja/coinkeeper/model/db/converter/UriConverter.java
package com.coinninja.coinkeeper.model.db.converter;

import android.net.Uri;

import org.greenrobot.greendao.converter.PropertyConverter;

public class UriConverter implements PropertyConverter<Uri, String> {

    @Override
    public Uri convertToEntityProperty(String databaseValue) {
        if (databaseValue == null || databaseValue.isEmpty()) {
            return null;
        }
        return Uri.parse(databaseValue);
    }

    @Override
    public String convertToDatabaseValue(Uri entityProperty) {
        if (entityProperty == null || entityProperty.toString().isEmpty()) {
            return null;
        }
        return entityProperty.toString();
    }
}

File: persistance/src/main/java/app/coinninja/cn/persistance/converter/UriConverter.kt
package app.coinninja.cn.persistance.converter

import android.net.Uri
import androidx.room.TypeConverter

object UriConverter {

    @TypeConverter
    @JvmStatic
    fun toUri(value: String): Uri {
        return Uri.parse(value)
    }

    @TypeConverter
    @JvmStatic
    fun toValue(uri: Uri): String {
        return uri.toString()
    }
}

